components/Categories.php

```diff
@@ -103,8 +103,15 @@ protected function loadCategories()
 
     protected function linkCategories($categories)
     {
-        return $categories->each(function($category) {
-            $category->setUrl($this->categoryPage, $this->controller);
+        $blogPostsComponent = $this->getComponent('blogPosts', $this->categoryPage);
+
+        return $categories->each(function($category) use ($blogPostsComponent) {
```

Comment:
@GinoPane Please add a space between `function` and the arguments as per PSR-2 guidelines.
```suggestion
        return $categories->each(function ($category) use ($blogPostsComponent) {
```